frontend/src/metabase-lib/compat/compat.ts

```diff
@@ -0,0 +1,96 @@
+import * as ML from "metabase-lib";
+import type { FieldFilter } from "metabase-types/api";
+import type StructuredQuery from "metabase-lib/queries/StructuredQuery";
+import LegacyFilter from "metabase-lib/queries/structured/Filter";
+
+export const getOperatorsMap = ({
+  query,
+  column,
+  stageIndex = -1,
+}: {
+  query?: ML.Query;
+  column?: ML.ColumnWithOperators;
+  stageIndex?: number;
+}) => {
+  if (!query || !column) {
+    return {};
+  }
+
+  const operators = ML.filterableColumnOperators(column);
+  return Object.fromEntries(
+    operators.map(operator => [
+      ML.displayInfo(query, stageIndex, operator).shortName,
+      operator,
+    ]),
+  );
+};
+
+export const getMlv2FilterClause = (filter: LegacyFilter, stageIndex = -1) => {
+  const mlv2Query = filter.query().question()._getMLv2Query();
+
+  const appliedFilterClause = ML.findFilterForLegacyFilter(
+    mlv2Query,
+    stageIndex,
+    [...(filter as unknown as FieldFilter)],
+  );
+
+  const column = appliedFilterClause
+    ? ML.filterParts(mlv2Query, stageIndex, appliedFilterClause).column
+    : ML.findFilterableColumnForLegacyRef(mlv2Query, stageIndex, filter[1]);
+
+  if (!column) {
+    throw new Error(`Could not find column for filter ${filter}`);
+  }
+
+  const operatorsMap = getOperatorsMap({
+    query: mlv2Query,
+    column,
+    stageIndex,
+  });
+
+  const unappliedFilterClause = ML.filterClause(
+    operatorsMap[filter[0]],
+    column as ML.ColumnMetadata,
+    ...filter.slice(2),
+  );
```

Comment:
Let's use filter methods to access filter properties instead of taking directly from MBQL:

```ts
const operator = operatorsMap[filter.operatorName()]
const args = filter.arguments()

ML.filterClause(operator, column, args);
```

Reply:
Also, why do we need to do `column as ML.ColumnMetadata` ?